sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -156,9 +189,10 @@ object ParquetReadSupport {
   def clipParquetSchema(
       parquetSchema: MessageType,
       catalystSchema: StructType,
-      caseSensitive: Boolean = true): MessageType = {
+      caseSensitive: Boolean,
```

Comment:
is the API break allowed here ?
Maybe it is better to add overloaded method ?!

Reply:
I did a code search and didn't find any breaking usages. I think the APIs here are private within Spark?
If we are talking about maintaining API compatibility for all `def`s in this class, we'd have to overload a bunch of other methods too bc they all had this additional `useFieldId` field.

@sadikovi what do you think? should we just make `useFieldId` default to false?

Reply:
I have not reviewed the newer changes yet but you can create an overloaded method without useFieldId to make sure we maintain binary compatibility.